src/plugins/dashboard/public/application/listing/dashboard_unsaved_listing.tsx

```diff
@@ -0,0 +1,161 @@
+import { DashboardAppServices, DashboardRedirect } from '../types';
+import { confirmDiscardUnsavedChanges } from './confirm_overlays';
+
+const DashboardUnsavedItem = ({
+  dashboard,
+  onOpenClick,
+  onDiscardClick,
+}: {
+  dashboard?: DashboardSavedObject;
+  onOpenClick: () => void;
+  onDiscardClick: () => void;
+}) => {
+  const title = dashboard?.title ?? getNewDashboardTitle();
+  return (
+    <div className="dshUnsavedListingItem">
+      <EuiFlexGroup alignItems="center" gutterSize="none">
+        <EuiFlexItem grow={false}>
+          <EuiIcon color="text" className="dshUnsavedListingItem__icon" type="dashboardApp" />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiTitle size="xxs">
+            <h4 className="dshUnsavedListingItem__title">
+              {dashboard?.title ?? getNewDashboardTitle()}
+            </h4>
+          </EuiTitle>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup
+        gutterSize="none"
+        alignItems="flexStart"
+        className="dshUnsavedListingItem__actions"
+      >
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            flush="left"
+            size="s"
+            color="primary"
+            onClick={onOpenClick}
+            data-test-subj={`edit-unsaved-${title.split(' ').join('-')}`}
+            aria-label={dashboardUnsavedListingStrings.getEditAriaLabel(title)}
+          >
+            {dashboardUnsavedListingStrings.getEditTitle()}
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            size="s"
+            color="danger"
+            onClick={onDiscardClick}
+            data-test-subj={`discard-unsaved-${title.split(' ').join('-')}`}
+            aria-label={dashboardUnsavedListingStrings.getDiscardAriaLabel(title)}
+          >
+            {dashboardUnsavedListingStrings.getDiscardTitle()}
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </div>
+  );
+};
+
+export const DashboardUnsavedListing = ({ redirectTo }: { redirectTo: DashboardRedirect }) => {
+  const {
+    services: {
+      dashboardPanelStorage,
+      savedDashboards,
+      core: { overlays },
+    },
+  } = useKibana<DashboardAppServices>();
+
+  const [items, setItems] = useState<JSX.Element[]>([]);
+  const [dashboardIds, setDashboardIds] = useState<string[]>([]);
+
+  const onOpen = useCallback(
+    (id?: string) => {
+      redirectTo({ destination: 'dashboard', id, editMode: true });
+    },
+    [redirectTo]
+  );
+
+  const onDiscard = useCallback(
+    (id?: string) => {
+      confirmDiscardUnsavedChanges(overlays, () => {
+        dashboardPanelStorage.clearPanels(id);
+        setDashboardIds(dashboardPanelStorage.getDashboardIdsWithUnsavedChanges());
+      });
+    },
+    [overlays, dashboardPanelStorage]
+  );
+
+  useMount(() => {
+    setDashboardIds(dashboardPanelStorage.getDashboardIdsWithUnsavedChanges());
```

Comment:
nit: looks like this should just be an initial state inside `useState`.

```
const [dashboardIds, setDashboardIds] = useState<string[]>(() => dashboardPanelStorage.getDashboardIdsWithUnsavedChanges());
```

Reply:
Much simpler, fixed this!